components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -268,6 +268,20 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             }
         }
 
+        String loginHint = request.getParameter(BasicAuthenticatorConstants.LOGIN_HINT);
+
+        if (StringUtils.isNotBlank(loginHint)) {
+            String inputType = FrameworkConstants.INPUT_TYPE_LOGIN_HINT;
+            if (StringUtils.isNotBlank(inputType)) {
```

Comment:
Do we need this blank check? `LOGIN_HINT` is a constant and feel redundant. Also no need to introduce a separate variable as well